### DISCUSSION TITLE
June 25 - Literate Programming I - Writing

### DISCUSSION THREAD

**Discussion by tonyfast** [Quansight/writers-workshop#18] [New class]
# the goal of the workshop

* [begin our journey into literate programming by talking about writing](https://nbviewer.jupyter.org/github/tonyfast/syllabus/blob/main/docs/i.md.ipynb)

## teacher's preferred ground rules

* How to use the raise handle feature? Keeping the professor on track..
* I encourage y'all to interrupt me when I'm talking or demo'ing. 
  Sometimes I get distracted by my own special interests and this class if for you all.
* To help take notes, please respond to this discussion item and continue editing your one reply during class. i can see the changes on the discussion board as they happen.

## [call to action] :+1: :-1:  :smile: :confused:  :rocket:  :heart:  :eyes: :tada: attendance

react to this discussion so we can take attendance.

![image](https://user-images.githubusercontent.com/4236275/117461813-fd163180-af1b-11eb-901b-bfdee65d8bdc.png)

### Q1 how's your energy

* :+1: i feel energetic from this week
* :-1: i feel exhausted from this week

### Q2 how do you want to participate

* :heart:   i'm happy to be at the writers workshop this week
* :eyes: i want to just watch and listen
* :tada: i am okay being called on for discussions

## note taking

please use this discussion as a place to take notes. create a comment that you edit throughout the lesson. 

_the perfect is the enemy of the good_

## agenda

0. [:00-:07] chit chat
1. show and tell
1. writing for fun and pleasure
2. our first introduction into literate programs.
3. writing for ourselves.
4. bounding ideas and time
5. homeworks
4. [:50-:60] wrapup

write! right?

**Comment by tonyfast** [comment]
https://github.com/MarsBarLee/writers-workshop/blob/fb4c46f597ebe69438b28d559516eb3ef31e3e62/Writer's%20Workshop%20%233-%20This%20Little%20Piggy....ipynb

![image](https://user-images.githubusercontent.com/4236275/123449802-02057200-d5aa-11eb-9113-22c3819ed02d.png)

## why is harding hard?

* getting started is hard
* i need a lot of time to do good writing
* organizing thoughts is hard
* assignments are hard and unmotivating, expectations take out the fun
* words are hard
* finishing is hard
* english is non-primary language
* not learning about writing in school

**Reply by MarsBarLee** [reply]
[NB Viewer version ](https://nbviewer.jupyter.org/github/MarsBarLee/writers-workshop/blob/MarsBarLee-submission/writer-workshop-3-mars-lee.ipynb) of the notebook

**Comment by MarsBarLee** [comment]
Literate Programming:

- Using markdown to bring meaning to code
- combat 'empty page syndrome' by 1) bounding your ideas, 2) bounding your audience, 3) bounding your time
1)  'restart and run all' or it didn't happen
2) Write for yourself -> class -> Quansight ->whoever, whatever
3) maybe putting time in your calendar

**This week's writing prompt**
Write a literate program, write about code like it's a subject

**Comment by tonyfast** [comment]
this part of the course needs to:

*  include more women
  * consider barbara kruger, jane jacobs, toni morrison, annette vee
*  discuss outlining
* omit some outline stuff that wasn't used
* describe different writing interfaces

for the exercises, we will have a few notebooks containing only code, and folks can practice adding narrative to the code. examples will include python, css, html, markdown.

![](https://dmdlnu87i51n1.cloudfront.net/v1/uk/ckq9etsz4000o3h5qljy0mkur/0x0:1600x849/undefinedxundefined/336_ex_jf_kruger_bk_forever3_press_yes_rt.jpg) https://www.theartnewspaper.com/interview/barbara-kruger-thank-god-i-m-an-artist
